mm0-rs/src/util.rs

```diff
@@ -268,6 +269,12 @@ fn get_memory_rusage() -> usize {
 
 /// Try to get total memory usage (stack + data) in bytes using the `/proc` filesystem.
 /// Falls back on `getrusage()` if procfs doesn't exist.
-pub(crate) fn get_memory_usage() -> usize {
-  procinfo::pid::statm_self().map_or_else(|_| get_memory_rusage(), |stat| stat.data * 4096)
+#[cfg(target_os = "linux")]
+pub(crate) fn get_memory_usage() -> Option<usize> {
```

Comment:
The `get_memory_rusage` function is already being used on linuxes where procfs doesn't exist; you could have `get_memory_usage` redirect to `get_memory_rusage` directly on non-linuxes. (I think it is still possible that `get_memory_rusage` can return 0, but it's in libc so it's pretty portable.)